example/example-model/src/main/java/uk/gov/gchq/palisade/example/rule/BankDetailsRule.java

```
@@ -41,14 +44,32 @@ public Employee apply(final Employee record, final User user, final Context cont
         if (null == record) {
             return null;
         }
-
         requireNonNull(user);
         requireNonNull(context);
         Set<String> roles = user.getRoles();
         String purpose = context.getPurpose();
 
-        if (roles.contains(Role.PAYROLL.name()) & purpose.equals(Purpose.SALARY.name())) {
-            return record;
+        System.out.println("class type");
+        System.out.println(user.getClass());
+
+
+        try {
+
+            if (user.getClass() == ExampleUser.class) {
+                ExampleUser exampleUser = (ExampleUser) user;
+                if (roles.contains(Role.PAYROLL.name()) && purpose.equals(Purpose.SALARY.name())) {
+                    return record;
+                }
+                EnumSet<TrainingCourse> trainingCompleted = exampleUser.getTrainingCompleted();
+                if (trainingCompleted.contains(TrainingCourse.PAYROLL_TRAINING_COURSE) & purpose.equals(Purpose.SALARY.name())) {
+                    return record;
+                }
+            }
+        } catch (Exception e) {
```

Remove try/catch. Nothing in this code should throw an exception.